mbtools/generate_mbz_toc.py

```diff
@@ -48,17 +49,67 @@ def create_toc(mbz_path, md_filepath):
                             current_page = id2page[current_page.next]
 
 
+def create_toc_csv(mbz_path, csv_filepath):
+    moodle_backup = MoodleBackup(mbz_path)
+    activity_list = []
+    for section in moodle_backup.sections():
+        for act in moodle_backup.activities(section.id):
+            if isinstance(act, MoodlePage):
+                for html_elem in act.html_elements():
+                    page_dict = {}
+                    page_dict['section'] = section.title
+                    page_dict['content_id'] = html_elem.get_attribute_values(
+                        "data-content-id")[0]
+                    page_dict['activity_name'] = act.name
+                    page_dict['lesson_page'] = ""
+
+                    activity_list.append(page_dict)
+            elif isinstance(act, MoodleLesson):
+                id2page = {}
+                current_page = None
+                for page in act.lesson_pages():
+                    id2page[page.id] = page
+                    if (page.prev == "0"):
+                        current_page = page
+                while (True):
+                    html_elem = current_page.html_element()
+                    lesson_dict = {}
+                    lesson_dict['section'] = section.title
+                    lesson_dict['content_id'] = html_elem.get_attribute_values(
+                        "data-content-id")[0]
+                    lesson_dict['activity_name'] = act.name
+                    lesson_dict['lesson_page'] = current_page.name
+                    activity_list.append(lesson_dict)
+                    if (current_page.next == "0"):
+                        break
+                    else:
+                        current_page = id2page[current_page.next]
+
+    with open(f"{csv_filepath}/toc.csv", "w") as outfile:
+        content_headers = activity_list[0].keys()
+        result = csv.DictWriter(
+            outfile,
+            fieldnames=content_headers)
+        result.writeheader()
+        result.writerows(activity_list)
+
+
 def main():
     parser = argparse.ArgumentParser(description='')
     parser.add_argument('mbz_path', type=str,
                         help='relative path to the mbz directory')
-    parser.add_argument('md_filepath', type=str,
+    parser.add_argument('output_path', type=str,
                         help='Path/name of the markdown file to be generated')
+    parser.add_argument('--csv', action='store_true',
+                        help="Generate a TOC CSV file")
+
     args = parser.parse_args()
     mbz_path = Path(args.mbz_path).resolve(strict=True)
-    md_filepath = args.md_filepath
-
-    create_toc(mbz_path, md_filepath)
+    output_path = args.output_path
+    csv = args.csv
+    create_toc_md(mbz_path, output_path)
+    if csv:
+        create_toc_csv(mbz_path, output_path)
```

Comment:
I feel like this should invoke one method or the other depending on the value of `csv`. Right now it will generate both of `--csv` is provided. Having it be one or the either might help as are as not hard coding a file name (this PR currently makes a significant change in that it now expects a user to provide a path + longer allows the caller to specify the output fail name).